smk/workflow/scripts/4_1_filter_open_targets_to_high_quality_gwas.py

```diff
@@ -0,0 +1,20 @@
+import sys
+sys.path.insert(1, snakemake.config['repodir'])
+from src.evaluation import create_opentargets_gwas_benchmarking_data
+
+import logging
+logging.basicConfig(filename=snakemake.log[0],
+                    level = logging.INFO)
+    
```

Comment:
@aron0093 Question about snakemake for you:

I had snakemake working for this script and the associated rule when I put my `filter_open_targets_gwas_query()` function in this  `smk/workflow/scripts` file, but it seemed more appropriate for me to move all of these functions into `src/evaluation/create_opentargets_gwas_benchmarking_data.py`.

Now, even though I have imported the `src.evaluation` module, I am not able to run `snakemake -s 4_1_filter_open_targets_to_high_quality_gwas.smk --cores 1`, although it ran successfully before I moved the `filter_open_targets_gwas_query()` in `src.evaluation` instead of in this file. Is it clear to you why? Have I not imported it properly?

Reply:
Error looks like this, although I have ensured that the `repodir` parameter is set in the config.yaml file.

```
[Wed Apr 10 14:07:31 2024]
rule run_filter_opentargets_query:
    input: ../../resources/OpenTargets_L2G_noQC.csv.gz
    output: ../../resources/OpenTargets_L2G_Filtered.csv.gz
    log: ../../logs/4_1_run_filter_OpenTargets_Query.log
    jobid: 0
    reason: Missing output files: ../../resources/OpenTargets_L2G_Filtered.csv.gz
    resources: tmpdir=/scratch/lsftmp/5276852.tmpdir

**Traceback (most recent call last):
  File "/lila/home/robertg1/gene_program_evaluation/smk/workflow/rules/.snakemake/scripts/tmpnytqs8uz.4_1_filter_open_targets_to_high_quality_gwas.py", line 6, in <module>
    sys.path.insert(1, snakemake.config['repodir'])
KeyError: 'repodir'**

[Wed Apr 10 14:07:31 2024]
Error in rule run_filter_opentargets_query:
    jobid: 0
    input: ../../resources/OpenTargets_L2G_noQC.csv.gz
    output: ../../resources/OpenTargets_L2G_Filtered.csv.gz
    log: ../../logs/4_1_run_filter_OpenTargets_Query.log (check log file(s) for error details)

RuleException:
CalledProcessError in file /lila/home/robertg1/gene_program_evaluation/smk/workflow/rules/4_1_filter_open_targets_to_high_quality_gwas.smk, line 11:
Command 'set -euo pipefail;  /home/robertg1/miniconda3/envs/cell_programs/bin/python /lila/home/robertg1/gene_program_evaluation/smk/workflow/rules/.snakemake/scripts/tmpnytqs8uz.4_1_filter_open_targets_to_high_quality_gwas.py' returned non-zero exit status 1.
  File "/lila/home/robertg1/gene_program_evaluation/smk/workflow/rules/4_1_filter_open_targets_to_high_quality_gwas.smk", line 11, in __rule_run_filter_opentargets_query
  File "/home/robertg1/miniconda3/envs/cell_programs/lib/python3.9/concurrent/futures/thread.py", line 58, in run
Shutting down, this might take some time.
Exiting because a job execution failed. Look above for error message
```

Reply:
oh, I see now that specifying the config file fixes this error (i.e. `snakemake -s 4_1_filter_open_targets_to_high_quality_gwas.smk --cores 1 --latency-wait 60 --configfile ~/gene_program_evaluation/smk/config/config.yaml`) 

However, then I get a different error:
```
Traceback (most recent call last):
  File "/lila/home/robertg1/gene_program_evaluation/smk/workflow/rules/.snakemake/scripts/tmp1tmfo8a8.4_1_filter_open_targets_to_high_quality_gwas.py", line 7, in <module>
    from src.evaluation import create_opentargets_gwas_benchmarking_data
ModuleNotFoundError: No module named 'src'
```

Reply:
Have you set 'repodir' in the config file? Unless the package is installed, it won't be in your PATH by default. 

`import sys
sys.path.insert(1, snakemake.config['repodir'])
`

Reply:
Now I think I have fixed that problem (I think snakemake was not referencing input and output files properly from the working directory), but I have a new one:
```
(cell_programs) [robertg1@lt20 rules]$ snakemake -s 4_1_filter_open_targets_to_high_quality_gwas.smk --cores 1 --latency-wait 60 --configfile /home/robertg1/gene_program_evaluation/smk/config/config.yaml 
Building DAG of jobs...
Using shell: /usr/bin/bash
Provided cores: 1 (use --cores to define parallelism)
Rules claiming more threads will be scaled down.
Job stats:
job                             count
----------------------------  -------
run_filter_opentargets_query        1
total                               1

Select jobs to execute...

[Wed Apr 10 15:16:18 2024]
rule run_filter_opentargets_query:
    input: ../../resources/OpenTargets_L2G_noQC.csv.gz
    output: ../../resources/OpenTargets_L2G_Filtered.csv.gz
    log: ../../logs/4_1_run_filter_OpenTargets_Query.log
    jobid: 0
    reason: Missing output files: ../../resources/OpenTargets_L2G_Filtered.csv.gz
    resources: tmpdir=/scratch/lsftmp/5276958.tmpdir

Error in sys.excepthook:

Original exception was:
[Wed Apr 10 15:16:28 2024]
Error in rule run_filter_opentargets_query:
    jobid: 0
    input: ../../resources/OpenTargets_L2G_noQC.csv.gz
    output: ../../resources/OpenTargets_L2G_Filtered.csv.gz
    log: ../../logs/4_1_run_filter_OpenTargets_Query.log (check log file(s) for error details)

RuleException:
CalledProcessError in file /lila/home/robertg1/gene_program_evaluation/smk/workflow/rules/4_1_filter_open_targets_to_high_quality_gwas.smk, line 11:
Command 'set -euo pipefail;  /home/robertg1/miniconda3/envs/cell_programs/bin/python /lila/home/robertg1/gene_program_evaluation/smk/workflow/rules/.snakemake/scripts/tmpczj1tb_y.4_1_filter_open_targets_to_high_quality_gwas.py' returned non-zero exit status 1.
  File "/lila/home/robertg1/gene_program_evaluation/smk/workflow/rules/4_1_filter_open_targets_to_high_quality_gwas.smk", line 11, in __rule_run_filter_opentargets_query
  File "/home/robertg1/miniconda3/envs/cell_programs/lib/python3.9/concurrent/futures/thread.py", line 58, in run
Shutting down, this might take some time.
Exiting because a job execution failed. Look above for error message
Complete log: .snakemake/log/2024-04-10T151618.351508.snakemake.log
```

Reply:
> Have you set 'repodir' in the config file? Unless the package is installed, it won't be in your PATH by default.
> 
> `import sys sys.path.insert(1, snakemake.config['repodir']) `

You need to import functions like this

`from src.evaluation import filter_open_targets_gwas_query`

instead of 

`from src.evaluation import create_opentargets_gwas_benchmarking_data`

Reply:
Additionally the rules will reference paths like this,

```
    input: resources/OpenTargets_L2G_noQC.csv.gz
    output: resources/OpenTargets_L2G_Filtered.csv.gz
    log: logs/4_1_run_filter_OpenTargets_Query.log
```

instead of 

```
    input: ../../resources/OpenTargets_L2G_noQC.csv.gz
    output: ../../resources/OpenTargets_L2G_Filtered.csv.gz
    log: ../../logs/4_1_run_filter_OpenTargets_Query.log
```

the paths to the scripts are fine `'../scripts/...`

Reply:
Unfortunately I am still getting an error after pulling in these fixes:
```
(cell_programs) [robertg1@lt14 rules]$ snakemake -s 4_run_query_open_targets_analysis.smk --cores 1 --latency-wait 60 --configfile ~/gene_program_evaluation/smk/config/config.yaml

SyntaxError in file /lila/home/robertg1/gene_program_evaluation/smk/workflow/rules/4_run_query_open_targets_analysis.smk, line 4:
invalid syntax
```

Reply:
I see now.  It just needed square brackets instead of curly ones (i.e. input: `config[credentials']`).  The first step runs now, but I see that you've combined the two rules into one script.  The first rule will run, but not the second.

Reply:
I have figured this out too now.  If I add this to the top of the .smk rrule file, both steps will run automatically:
`rule all:
  input:
      'resources/OpenTargets_L2G_Filtered.csv.gz'`

Reply:
Sorry. Once I had finished editing stuff the code was all over the place so I decided to pull a fresh copy and make the final edits. Should have run it before pushing.